frappe/utils/background_jobs.py

```diff
@@ -18,12 +18,17 @@
 from frappe import _
 from frappe.utils import cstr
 
+site_config = frappe.get_site_config()
+common_site_config = frappe.get_file_json("common_site_config.json")
+custom_workers_config = site_config.get("workers", common_site_config.get("workers", {}))
 default_timeout = 300
-queue_timeout = {
-	'long': 1500,
-	'default': 300,
-	'short': 300
-}
+queue_timeout = dict(
+	{"default": default_timeout, "short": default_timeout, "long": 1500},
+	**{
+		worker_name: worker_config.get("timeout", default_timeout)
+		for worker_name, worker_config in custom_workers_config.items()
+	},
+)
```

Comment:
This feels a bit congested. I think we can spread them out a bit?

```suggestion
queue_timeout = {
	"default": default_timeout,
	"short": default_timeout,
	"long": 1500,
}
queue_timeout.update({
	worker: config.get("timeout", default_timeout)
	for worker, config in custom_workers_config.items()
})
```

Reply:
:+1: